### PR TITLE
fix Meson default cross-build x64->32

### LINKED ISSUE #17261: [question] Why do we skip 64/32 cross compilation with meson?
See https://github.com/conan-io/conan/blob/d35a73142a538808083f775ba3108664e1d50415/conan/tools/meson/toolchain.py#L159C2-L159C73

We are trying to cross build glib for linux x64->x86 (`arch=x86`) using the official conancenter recipe and the host profile
```
[settings]
os=Linux
arch=x86
compiler=clang
compiler.version=14
compiler.libcxx=libstdc++11
build_type=Debug
```

This leads to the following error:
`/usr/bin/ld: i386 architecture of input file '/conan/.conan/data/pcre2/10.42/_/_/package/9ff55d811dff94adb5ea6675dae30929f0465371/lib/libpcre2-8.a(pcre2_chartables.c.o)' is incompatible with i386:x86-64 output`
see [glib_error_logs.txt](https://github.com/user-attachments/files/17588545/glib_error_logs.txt)

It can be fixed by setting `tools.build.cross_building:cross_build=True`, however shouldn't this work out of the box without any special configuration?

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Fix: Meson aligns with other build systems considering `x86_64`->`x86` as cross building.
Docs: Omit

Close https://github.com/conan-io/conan/issues/17261

### CODE REVIEW

File: conan/tools/meson/toolchain.py [hunk 1]
```diff
@@ -424,13 +424,13 @@ def _get_extra_flags(self):
         exelinkflags = self._conanfile_conf.get("tools.build:exelinkflags", default=[], check_type=list)
         linker_scripts = self._conanfile_conf.get("tools.build:linker_scripts", default=[], check_type=list)
         linker_script_flags = ['-T"' + linker_script + '"' for linker_script in linker_scripts]
-        defines = self._conanfile.conf.get("tools.build:defines", default=[], check_type=list)
+        defines = self._conanfile_conf.get("tools.build:defines", default=[], check_type=list)
```
Comment (memsharded) [inline conan/tools/meson/toolchain.py:427]: Check this line @franramirez688, was it a bug or intended?

File: conan/tools/meson/toolchain.py [hunk 2]
```diff
@@ -156,7 +156,7 @@ def __init__(self, conanfile, backend=None, native=False):
         self._conanfile = conanfile
         self._native = native
         self._is_apple_system = is_apple_os(self._conanfile)
-        is_cross_building = cross_building(conanfile, skip_x64_x86=True)
+        is_cross_building = cross_building(conanfile)  # x86_64->x86 is considered cross-building
```
Comment (memsharded) [inline conan/tools/meson/toolchain.py:159]: This is the bug fix

File: conan/tools/meson/toolchain.py (hunk 1 above)
Comment (franramirez688) [reply conan/tools/meson/toolchain.py:427]: It's a bug. I don't know why I left this. I just reviewed the code again and I saw [the lines above](https://github.com/conan-io/conan/pull/17266/files#diff-0e1e7a3c21f2c036f65b81f8dc72f539f9c76b2b0a031cd5eca0f1de392bd49eR167) this:

```python
self._backend = conanfile.conf.get("tools.meson.mesontoolchain:backend",
                                           default=backend or 'ninja')
```

Please, change it to:
```python
self._backend = self._conanfile_conf.get("tools.meson.mesontoolchain:backend",
                                           default=backend or 'ninja')
```

File: conan/tools/meson/toolchain.py [hunk 3]
```diff
@@ -447,11 +447,11 @@ def _filter_list_empty_fields(v):
     def _sanitize_env_format(value):
         if value is None or isinstance(value, list):
             return value
+        assert isinstance(value, str)
```
Comment (franramirez688) [inline conan/tools/meson/toolchain.py]: I think that we could raise a ConanException instead.

File: conan/tools/meson/toolchain.py (hunk 1 above)
Comment (memsharded) [reply conan/tools/meson/toolchain.py:427]: Done

File: conan/tools/meson/toolchain.py (hunk 3 above)
Comment (memsharded) [reply conan/tools/meson/toolchain.py]: Done

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**